FuncGodot Manual/pages/tips_runtime_building.html

```diff
@@ -39,11 +41,94 @@ <h2>Setting up Runtime map building in Godot</h2>
+        func_godot_map.verify_and_build()</code></pre>
+    <h2>Editor setup</h2>
+    <h3>TrenchBroom</h3>
+    <p>
+        The TrenchBroom documenation <a href="https://trenchbroom.github.io/manual/latest/#launching_game_engines">covers this</a>. You configure an engine you can run from which is its executable, and you pass a parameter to it, which is the map file. Trenchbroom provides variables for finding the path of the map file.
+    </p>
+
+    <h3>NetRadiant Custom</h3>
+
+    <p>
+        NetRadiant itself provides some basic scripting for this process, defined in the gamepack folder. 
+
+        The path to our default build profiles can be found in your NetRadiant gamepack folder. This is something defined in your 
+        <a href = https://func-godot.github.io/func_godot_docs/FuncGodot%20Manual/pages/ref_netradiant_custom_resources.html#GamepackConfig>NetRadiantCustomGamePackConfig</a>.
+
+        IF you haven't set this up, you'll need to first complete <a href = https://func-godot.github.io/func_godot_docs/FuncGodot%20Manual/pages/guide_map_editor_config.html#NetRadiantCustom>creating your NetRadiantCustomGamePackConfig.</a>
+
+        We'll use an example from Sinewave's game SLAMFIRE: <code class="highlight">&ltpath_to_radiant>\gamepacks\sfbase.game\default_build.xml</code>.
+    </p>
+    <p>
+
+    </p>
+        First, in Radiant, navigate to <code class="highlight">Build > Customize...</code>.
+
+        Here, we can see some build variables available:
+        <p align="center"><img src="../images/tips_editorbuild_radiant1.png"><br></p>
+
+        To create an entry, open up the default build profile (mentioned above). 
+        You can issue a command to execute depending on the currently running platform like so:
+
+        <code><pre>
+&lt?xml version="1.0"?>
+&ltproject version="2.0">
+&ltvar name="game">"[EnginePath]sf.[ExecutableType]"&lt/var>
+&ltbuild name="Test map in engine">
+&ltcommand>[game] -- launch_debug::"[MapFile]"&lt/command>
+&lt/build></code></pre>
+
+        Restart NetRadiant, and you should see a new option in the build menu.
+        <p align="center"><img src="../images/tips_editorbuild_radiant2.png"><br></p>
+
+        <p align="center"><img src="../images/tips_editorbuild_radiant3.png"><br></p>
+
+        Selecting your new option will run the executable you defined above!
     </p>
-    <h2>Using command lines to build from Map Editors</h2>
-    <p>Combined with <code>OS.get_cmdline_user_args();</code> and your map editor's build profile, you can open up a map you are editing and build it at runtime to test it. For Trenchbroom, this involves <a href="https://trenchbroom.github.io/manual/latest/#launching_game_engines">configuring launch options</a>.</p>
+    <blockquote><i>NOTE: Where an executable is listed, you can provide the path to Godot Game engine with a parameter of the path to your Godot project. Alternatively you can also use exported executables.<br></i></blockquote>
```

Comment:
NOTE: For your map editor's game path, you may provide either the [Godot editor path](https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-property-editor-run-main-run-args), or an exported project's path.


Might need to look into syntax for using this with the editor proper; 
depending on how you define the project to launch in the editor, it's arguably easier to just export a build.

Reply:
For trenchbroom it's as simple as this:
`/home/odst/Programming/godot4/PerspectiveGame/project.godot ${MAP_BASE_NAME}` as the parameter
with my launch configuration executable set as where godot is(or my own compiled bin of godot)

See this as an example:
![image](https://github.com/func-godot/func_godot_docs/assets/16968644/dd9bfc89-f4a0-465e-ab10-78f19c54a472)

Reply:
> For trenchbroom it's as simple as this:
/home/odst/Programming/godot4/PerspectiveGame/project.godot ${MAP_BASE_NAME} as the parameter
with my launch configuration executable set as where godot is(or my own compiled bin of godot)

It isn't that simple if your .godot project doesn't open the Godot Editor. I think you may have changed that at some point to auto launch with Godot? For me it opens up VSCode.

Since it's not a default way to open up project.godot files we shouldn't put that in the tutorial proper. Maybe as block Note.

Reply:
> NOTE: For your map editor's game path, you may provide either the [Godot editor path with parameters to your project](https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-property-editor-run-main-run-args), or with an exported project's executable path.

Can either of you look into this? I'm admittedly fairly busy right now so I can act as a reviewer but not really a researcher atm. This is how we should be teaching it, since, as I just mentioned, opening the project.godot file does not open up Godot by default.

Reply:
I run on Linux. The idea here, is that you define a launch configuration which is what binary/executable is run. On Linux these binaries are found in /usr/bin/godot when installing from package manager. On windows, this should just be wherever your godot .exe is.

In a command line it is equivalent to running `path/to/your/godot.exe path/to/your/project.godot`. The reason it is opening up in vscode by default is because your system has .godot files associated with vscode.

Trenchbroom lets you set the launch configuration for the executable. Then when you launch, with that configuration you can pass parameters(i.e. project.godot as the first parameter as it's the project and your map these are your CLI args)

Reply:
> The reason it is opening up in vscode by default is because your system has .godot files associated with vscode.

I'm aware of this. But project.godot files don't open up the Godot Editor by default either. Not on Windows anyway, which has a much larger share of our user base than Linux and MacOS.

> In a command line it is equivalent to running path/to/your/godot.exe path/to/your/project.godot.

So it is. Useful to know. And then command line arguments go after it starting with `--` or `++`, and then the `::` for is just personal syntax (which works fine for the tutorial, so long as we explain its purpose).

So it effectively becomes
- Executable set to Godot.exe
- Pass command line as `"path\to\project.godot" --cmd::arg`

This accurate?

Reply:
This is accurate. See the new and improved TrenchBroom section for an example of this. It is much more tutorialised, with documentation links for Godot's CLI params and TrenchBroom's built in variables